packages/@ionic/cli/src/commands/package/build.ts

```diff
@@ -267,13 +313,49 @@ This can be used only together with build type ${input('release')} for Android a
       throw new Error(`Build ${build.state}`);
     }
 
-    const url = await this.getDownloadUrl(appflowId, buildId, token);
-    if (!url.url) {
-      throw new Error('Missing URL in response');
+    for (const artifactType of artifactTypes) {
+      const url = await this.getDownloadUrl(appflowId, buildId, artifactType, token);
+
+      if (!url.url) {
+        throw new Error('Missing URL in response');
+      }
+
+      let customBuildFileName = '';
+
+      if (options['build-file-name']) {
+        this.env.log.warn(`The ${input('--build-file-name')} option has been deprecated. Please use ${input('--ipa-name')}, ${input('--apk-name')} or ${input('--{ artifact }-name')}.`);
+        customBuildFileName = await this.sanitizeString(options['build-file-name']);
+      } else {
+        customBuildFileName = await this.sanitizeString(options[`${artifactType.toLowerCase()}-name`]);
+      }
+
+      const filename = await this.downloadBuild(url.url, customBuildFileName);
+      this.env.log.ok(`Artifact downloaded: ${filename}`);
+    }
+  }
+
+  async sanitizeString(value: string | string[] | boolean | null | undefined): Promise<string> {
+
+    if (!value || typeof (value) !== 'string') {
+      return '';
     }
 
-    const filename = await this.downloadBuild(url.url, customBuildFileName);
-    this.env.log.ok(`Build completed: ${filename}`);
+    if (!fileUtils.isValidFileName(value)) {
+      throw new FatalException(`${strong(String(value))} is not a valid file name`);
+    }
+
+    return String(value);
+  }
+
+  async getDefaultArtifactType(platform: string): Promise<Array<string>> {
+    switch (platform) {
+      case 'ios':
+        return ['IPA', 'DSYM'];
+      case 'android':
+        return ['APK', 'AAB'];
```

Comment:
After some more thought I think we should default to `IPA` and `APK`, that way if they were using `--build-file-name` before everything will still work the same going forward.